File: python/example8_1.py
# Example 8.1 Probit Model of Economic Education
# Binomial Regression Model: Probit and Logit
# Probit Model using statsmodels package

import numpy as np
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf 

data = pd.read_csv('http://web.pdx.edu/~crkl/ceR/data/grade.txt',sep='\s+', nrows=32) 
data = data.dropna()
print(data.describe())

X = data[['GPA','TUCE','PSI']]
y = data['GRADE']
model1 = sm.Probit(y,sm.add_constant(X)).fit()
model1.summary()

model2 = sm.Probit.from_formula('GRADE ~ GPA + TUCE + PSI',data=data).fit()
model2.summary()

# same as model2
model3 = smf.probit('GRADE ~ GPA + TUCE + PSI', data=data).fit()
model3.summary()

# using GLM
model4 = smf.glm(formula='GRADE ~ GPA + TUCE + PSI', 
                family=sm.families.Binomial(link = sm.genmod.families.links.probit),
                data=data).fit()
model4.summary()

# probit model interpretation
model2.params
# predicted probability of each obs
model2.predict(data)
# marginal effects at the mean of each regressor
margeff = model2.get_margeff('mean')
margeff.summary()

File: R/example8_1.R
# Example 8.1 Probit Model of Economic Education
# Binomial Regression Model: Probit and Logit

setwd("C:/Course17/ceR/R")
data<-read.table("http://web.pdx.edu/~crkl/ceR/data/grade.txt",header=T,nrows=32)

gpa<-data$GPA
tuce<-data$TUCE
psi<-data$PSI
grade<-data$GRADE

model2<-glm(grade~gpa+tuce+psi,binomial(link=probit))  # link=probit
summary(model2)

# probit model interpretation
b2<-model2$coefficients
probility<-predict(model2,type="response")       # estimated probability

# marginal effects of an explanatory variable
x<-cbind(1,gpa,tuce,psi)
gpa_slopes<-dnorm(x%*%b2,mean=0,sd=1)*b2["gpa"]
tuce_slopes<-dnorm(x%*%b2,mean=0,sd=1)*b2["tuce"]
psi_slopes<-dnorm(x%*%b2,mean=0,sd=1)*b2["psi"]

cbind(grade,probility,gpa_slopes,tuce_slopes,psi_slopes)

rm(list=ls())
